lib/map-over-styles-and-transform-background-image-declarations.js

```diff
@@ -66,6 +66,37 @@ function mapOverStylesAndTransformAllBackgroundImageDeclarations(styles, cb) {
 
 		}
 
+		if(rule.type === 'keyframes') {
+			// Get keyframe from keyframes
+			rule.keyframes = transformMap(rule.keyframes, function(keyframe, keyframeIndex, keyframes) {
+				// Get declarations from keyframe
+				keyframe.declarations = transformMap(keyframe.declarations, function(declaration, declarationIndex, declarations) {
+					// Clone the declartion to keep it immutable
+					var transformedDeclaration = extend(true, {}, declaration);
+					transformedDeclaration = attachInfoToDeclaration(declarations, declarationIndex);
+
+					// background-image always has a url
+					if(transformedDeclaration.property === 'background-image') {
+						return cb(transformedDeclaration, declarationIndex, declarations);
+					}
+					// Background is a shorthand property so make sure `url()` is in there
+					else if(transformedDeclaration.property === 'background') {
+						var hasImageValue = spriterUtil.backgroundURLRegex.test(transformedDeclaration.value);
+
+						if(hasImageValue) {
+							return cb(transformedDeclaration, declarationIndex, declarations);
+						}
+					}
+
+					// Wrap in an object so that the declaration doesn't get interpreted
+					return {
+						'value': transformedDeclaration
+					};
+				});
+
+			});
+		}
+

```

Comment:
We should do something like this to reduce the repeated logic

```js
function transformDeclaration(declaration, declarationIndex, declarations) {
	// Clone the declaration to keep it immutable
	var transformedDeclaration = extend(true, {}, declaration);
	transformedDeclaration = attachInfoToDeclaration(declarations, declarationIndex);

	// background-image always has a url
	if(transformedDeclaration.property === 'background-image') {
		return cb(transformedDeclaration, declarationIndex, declarations);
	}
	// Background is a shorthand property so make sure `url()` is in there
	else if(transformedDeclaration.property === 'background') {
		var hasImageValue = spriterUtil.backgroundURLRegex.test(transformedDeclaration.value);

		if(hasImageValue) {
			return cb(transformedDeclaration, declarationIndex, declarations);
		}
	}

	// Wrap in an object so that the declaration doesn't get interpreted
	return {
		'value': transformedDeclaration
	};
}


if(rule.type === 'rule') {
	rule.declarations = transformMap(rule.declarations, transformDeclaration);
}
else if(rule.type === 'keyframes') {
	// Get keyframe from keyframes
	rule.keyframes = transformMap(rule.keyframes, function(keyframe, keyframeIndex, keyframes) {
		// Get declarations from keyframe
		keyframe.declarations = transformMap(keyframe.declarations, transformDeclaration);
	});
}
```